src/docfx/build/context/BuildScope.cs

```diff
@@ -52,6 +58,12 @@ public bool GetActualFileName(string fileName, out string actualFileName)
             return _fileNames.TryGetValue(fileName, out actualFileName);
         }
 
+        public Docset GetFallbackDocset(Docset docset)
+            => docset == Docset || IsFallbackDocset(docset) ? _fallbackDocset : null;
+
+        public bool IsFallbackDocset(Docset docset)
```

Comment:
is it possible to detect if a `file` is a fallback by checking `file.FilePath.Origin == Fallback` ?

Reply:
They are actually same, sometimes we don't have file, only docset